src/components/Dao/DaoProposalsPage.tsx

```diff
@@ -0,0 +1,152 @@
+import * as React from "react";
+import * as css from "./DaoProposalsPage.scss";
+import { Address, IDAOState, Proposal } from "@daostack/arc.js";
+import { BreadcrumbsItem } from "react-breadcrumbs-dynamic";
+import withSubscription, { ISubscriptionProps } from "components/Shared/withSubscription";
+import Loading from "components/Shared/Loading";
+import InfiniteScroll from "react-infinite-scroll-component";
+import { Observable } from "rxjs";
+import ProposalRow from "components/Proposal/ProposalRow";
+import { RouteComponentProps } from "react-router-dom";
+import { first } from "rxjs/operators";
+import { isAddress } from "lib/util";
+import moment from "moment-timezone";
+
+const PAGE_SIZE = 50;
+
+type SubscriptionData = Array<Proposal>;
+type IProps = IExternalProps & ISubscriptionProps<SubscriptionData>;
+type IExternalProps = {
+  daoState: IDAOState;
+  currentAccountAddress: Address;
+} & RouteComponentProps<any>;
+
+type OrderDirection = "asc" | "desc";
+
+
+const proposalsQuery = (dao: IDAOState, skip: number, titleSearch?: string): Observable<Array<Proposal>> => {
+  const filter: any = {
+  };
+
+  let orderDirection: OrderDirection = "asc";
+
+  if (!titleSearch) {
+    if (skip % PAGE_SIZE === 0) {
+      filter["closingAt_gt"] = moment().unix();
+    } else {
+      skip -= dao.numberOfQueuedProposals + dao.numberOfPreBoostedProposals + dao.numberOfBoostedProposals;
+      filter["closingAt_lt"] = moment().unix();
+    }
+  }
+
+  if (titleSearch?.trim()) {
+    orderDirection = "desc";
+    if (isAddress(titleSearch)) {
+      filter["proposer"] = titleSearch;
+    } else {
+      filter["title_contains"] = titleSearch;
+    }
+  }
+
+  return dao.dao.proposals({
+    where: filter,
+    orderBy: "closingAt",
+    orderDirection: orderDirection,
+    first: titleSearch ? undefined : PAGE_SIZE, // TEMPORARY UNTIL WE PASS "titleSearch" in line 143
```

Comment:
Why this string? Looks quite wired. Could you please explain this temporary logic?

Reply:
This is a temporary hack until I'll figure out a way to pass the `titleSearch` string into here, since it's necessary